app/scripts/modules/amazon/src/function/configure/Network.tsx

```diff
@@ -0,0 +1,247 @@
+import * as React from 'react';
+import { Option } from 'react-select';
+import { IPromise } from 'angular';
+import { Observable, Subject } from 'rxjs';
+import { forOwn } from 'lodash';
+
+import {
+  Application,
+  FormikFormField,
+  IWizardPageComponent,
+  IVpc,
+  ISubnet,
+  HelpField,
+  IAccount,
+  IRegion,
+  ReactSelectInput,
+  TetheredSelect,
+  SubnetReader,
+  ReactInjector,
+  ISecurityGroupsByAccountSourceData,
+} from '@spinnaker/core';
+import { FormikErrors, FormikProps } from 'formik';
+import { IAmazonFunctionUpsertCommand } from 'amazon/index';
+import { VpcReader } from 'amazon/vpc';
+
+export interface ISubnetOption {
+  subnetId: string;
+  vpcId: string;
+}
+
+export interface INetworkProps {
+  formik: FormikProps<IAmazonFunctionUpsertCommand>;
+  isNew?: boolean;
+  app: Application;
+}
+
+export interface INetworkState {
+  vpcOptions: Array<{}>;
+  accounts: IAccount[];
+  regions: IRegion[];
+  subnets: ISubnetOption[];
+  securityGroups: ISecurityGroupsByAccountSourceData;
+}
+
+export class Network extends React.Component<INetworkProps, INetworkState>
+  implements IWizardPageComponent<IAmazonFunctionUpsertCommand> {
+  constructor(props: INetworkProps) {
+    super(props);
+    this.getAllVpcs();
+    // this.getAllSubnets();
+  }
+
+  public state: INetworkState = {
+    vpcOptions: [],
+    accounts: null,
+    regions: [],
+    subnets: [],
+    securityGroups: null,
+  };
+  private props$ = new Subject<INetworkProps>();
+  private destroy$ = new Subject<void>();
+
+  private getAllVpcs(): void {
+    VpcReader.listVpcs().then(Vpcs => {
+      this.state.vpcOptions = Vpcs;
+    });
+  }
+
+  public validate(): FormikErrors<IAmazonFunctionUpsertCommand> {
```

Comment:
Does Formik require a validate function even if no validation is required? It seems weird to have this function, but I haven't worked wit Formik enough to know.

Reply:
I believe so. For example,  I was facing the following error: 
```
    Class 'Network' incorrectly implements interface 'IWizardPageComponent<IAmazonFunctionUpsertCommand>'.
  Property 'validate' is missing in type 'Network' but required in type 'IWizardPageComponent<IAmazonFunctionUpsertCommand>'.ts(2420)
```

Reply:
this requirement is a wizard page thing